service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java

```diff
@@ -138,6 +139,7 @@ private ResponseEntity<ApiDeleteControlledAwsResourceResult> deleteAwsResource(
     ControlledResource awsResource =
         controlledResourceMetadataManager.validateControlledResourceAndAction(
             userRequest, workspaceUuid, resourceUuid, SamControlledResourceActions.DELETE_ACTION);
+    workspaceService.validateWorkspaceAndContextState(workspaceUuid, CloudPlatform.AWS);
```

Comment:
Should we also check resource state here? I don't think it's entirely necessary since that should happen as part of the flight, but it saves launching a flight

Reply:
I don't think we need to optimize launching flights. It is not a big cost.